java.hints/src/org/netbeans/modules/java/hints/bugs/NPECheck.java

```diff
@@ -429,17 +430,37 @@ public static ErrorDescription returnNull(HintContext ctx) {
 
         TreePath method = ctx.getPath();
 
-        while (method != null && method.getLeaf().getKind() != Kind.METHOD && method.getLeaf().getKind() != Kind.CLASS) {
+        loop: while (method != null) {
+            switch (method.getLeaf().getKind()) {
+                case LAMBDA_EXPRESSION:
+                case METHOD:
+                // if we reach the enclosing class and still have not found
+                // the path to the method or lambda expression, then break.
+                case CLASS:
+                    break loop;
+            }
             method = method.getParentPath();
         }
 
-        if (method == null || method.getLeaf().getKind() != Kind.METHOD) return null;
+        if (method == null) return null;
+
+        CompilationInfo info = ctx.getInfo();
 
-        Element el = ctx.getInfo().getTrees().getElement(method);
+        Element el = null;
+        switch (method.getLeaf().getKind()) {
+            case LAMBDA_EXPRESSION:
+                TypeMirror functionalType = info.getTrees().getTypeMirror(method);
+                if (functionalType == null || functionalType.getKind() != TypeKind.DECLARED) return null;
```

Comment:
instead of == null, use Utilities.isTypeValid(). It returns false for various non-null, but erroneous types (i.e. half-written source produces strange things)

Reply:
Done